packages/backend/src/queryBuilder.ts

```diff
@@ -1,125 +1,133 @@
 import {
-    BooleanFilter,
     CompiledMetricQuery,
-    DateAndTimestampFilter,
+    DimensionType,
     Explore,
     fieldId,
     FieldId,
-    fieldIdFromFilterGroup,
-    FilterGroup,
-    FilterGroupOperator,
+    FilterOperator,
+    FilterRule,
     formatDate,
     formatTimestamp,
     getDimensions,
+    getFilterRulesFromGroup,
     getMetrics,
-    NumberFilter,
-    StringFilter,
+    getTotalFilterRules,
+    isAndFilterGroup,
     SupportedDbtAdapter,
 } from 'common';
 
 const renderStringFilterSql = (
     dimensionSql: string,
-    filter: StringFilter,
+    filter: FilterRule,
 ): string => {
     const filterType = filter.operator;
     switch (filter.operator) {
-        case 'equals':
-            return filter.values.length === 0
+        case FilterOperator.EQUALS:
+            return filter.values?.length === 0
                 ? 'false'
-                : `(${dimensionSql}) IN (${filter.values
+                : `(${dimensionSql}) IN (${(filter.values || [])
                       .map((v) => `'${v}'`)
                       .join(',')})`;
-        case 'notEquals':
-            return filter.values.length === 0
+        case FilterOperator.NOT_EQUALS:
+            return filter.values?.length === 0
                 ? 'true'
-                : `(${dimensionSql}) NOT IN (${filter.values
+                : `(${dimensionSql}) NOT IN (${(filter.values || [])
                       .map((v) => `'${v}'`)
                       .join(',')})`;
-        case 'doesNotInclude':
-            return `(${dimensionSql}) NOT LIKE '%${filter.value}%'`;
-        case 'isNull':
+        case FilterOperator.NOT_INCLUDE:
+            return `(${dimensionSql}) NOT LIKE '%${filter.values?.[0]}%'`;
+        case FilterOperator.NULL:
             return `(${dimensionSql}) IS NULL`;
-        case 'notNull':
+        case FilterOperator.NOT_NULL:
             return `(${dimensionSql}) IS NOT NULL`;
-        case 'startsWith':
-            return `(${dimensionSql}) LIKE '${filter.value}%'`;
+        case FilterOperator.STARTS_WITH:
+            return `(${dimensionSql}) LIKE '${filter.values?.[0]}%'`;
         default:
-            const nope: never = filter;
             throw Error(
-                `No function implemented to render sql for filter type ${filterType} on dimension of number type`,
+                `No function implemented to render sql for filter type ${filterType} on dimension of string type`,
             );
     }
 };
 
 const renderNumberFilterSql = (
     dimensionSql: string,
-    filter: NumberFilter,
+    filter: FilterRule,
 ): string => {
     const filterType = filter.operator;
     switch (filter.operator) {
-        case 'equals':
-            return filter.values.length === 0
+        case FilterOperator.EQUALS:
```

Comment:
I was able to reach this with a string argument:
<img width="573" alt="Screenshot 2022-01-31 at 16 02 39" src="https://user-images.githubusercontent.com/11660098/151830429-497d92b0-599f-44ad-9bbb-80896d50519c.png">
<img width="927" alt="Screenshot 2022-01-31 at 16 02 35" src="https://user-images.githubusercontent.com/11660098/151830432-548ecc35-6a2f-4233-bab5-de5b3110a0bd.png">